src/store/sagas/swap.ts

```diff
@@ -470,15 +473,13 @@ export function* handleGetSimulateResult(action: PayloadAction<Simulate>) {
           continue
         }
 
-        const calculatedAmountOut = byAmountIn ? result.amountOut : result.amountOut + result.fee
-
-        if (calculatedAmountOut === 0n) {
+        if (result.amountOut === 0n) {
           errors.push(SwapError.AmountIsZero)
           continue
         }
 
-        if (calculatedAmountOut > amountOut) {
-          amountOut = calculatedAmountOut
+        if (byAmountIn ? result.amountOut > amountOut : result.amountIn + result.fee > amountOut) {
```

Comment:
refactor this piece of code to make it more readable - handles routing (choosing the pool). It's possible that there is a bug here as well